packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -309,20 +309,66 @@ export class ProfilesUtils {
+                \nDo you wish to disable credential management for all setting scopes? (VS Code window reload will be triggered)`
+            ),
+            {
+                items: [yesButton],
+                vsCodeOpts: {
+                    modal: true,
+                },
+            }
+        );
+        if (response === yesButton) {
+            await SettingsConfig.setDirectValueForAll("zowe.security.secureCredentialsEnabled", false);
+            await vscode.commands.executeCommand("workbench.action.reloadWindow");
+        } else {
+            throw new imperative.ImperativeError({
+                msg: localize(
+                    "ProfileUtils.promptAndDisableCredentialManagement.error",
+                    `Failed to load credential manager. This may be related to Zowe Explorer being unable 
+                    to use the default credential manager in a browser based environment.`
+                ),
+            });
+        }
+    }
+
     /**
      * Use the default credential manager in Zowe Explorer and setup before use
      * @returns Promise<imperative.ProfileInfo> the object of profileInfo using the default credential manager
      */
     public static async setupDefaultCredentialManager(): Promise<imperative.ProfileInfo> {
-        ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
-        ZoweLogger.info(
-            localize("ProfilesUtils.setupDefaultCredentialManager.usingDefault", "No custom credential managers found, using the default instead.")
-        );
-        await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
-        return new imperative.ProfileInfo("zowe", {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-            credMgrOverride: imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring),
-        });
+        try {
+            ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
+            ZoweLogger.info(
+                localize(
+                    "ProfilesUtils.setupDefaultCredentialManager.usingDefault",
+                    "No custom credential managers found, using the default instead."
+                )
+            );
+            await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
+            const defaultCredentialManager = imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring);
+            const profileInfo = new imperative.ProfileInfo("zowe", {
+                credMgrOverride: defaultCredentialManager,
+            });
+            // Trigger initialize() function of credential manager to throw an error early if failed to load
+            await profileInfo.readProfilesFromDisk();
+            return profileInfo;
+        } catch (err) {
+            await ProfilesUtils.promptAndDisableCredentialManagement();
```

Comment:
I like the cleaner error messages from an end-user standpoint, but can we also log the error that occurs while initializing the credential manager? Otherwise, it might make it more difficult to find the source of the issue if the details are not accessible outside of this try/catch block.